src/test/java/org/owasp/html/EncodingTest.java

```diff
@@ -188,6 +188,15 @@ public static final void testDecodeHtml() {
     assertEquals(
         "\u03b1",
         Encoding.decodeHtml("&alpha;"));
+    assertEquals(
+        "\ud835\udc9c",  // U+1D49C requires a surrogate pair in UTF-16.
+        Encoding.decodeHtml("&Ascr;"));
+    assertEquals(
+        "fj",  // &fjlig; refers to 2 characters.
+        Encoding.decodeHtml("&fjlig;"));
+    assertEquals(
+        "\u2233",  // HTML entity with the longest name.
+        Encoding.decodeHtml("&CounterClockwiseContourIntegral;"));
```

Comment:
Cool.  Maybe suss out IndexOutOfBounds via

```js
   assertEquals(
        // HTML entity with the longest name missing terminator.
        "&CounterClockwiseContourIntegral",  
        Encoding.decodeHtml("&CounterClockwiseContourIntegral"));
```

Reply:
Sorry, I'm not sure I follow. Can you explain the case you want to protect against here?

Incidentally that input string does get decoded to `∳` because there is some logic for detecting "broken entities" that end with a non-alphanumeric character or end of string.

Reply:
Since that's the longest, it's the most likely culprit in any boundary bug that leads to attempts to read past the end of input looking for a semicolon or another character in a character reference name.

> Incidentally that input string does get decoded to ∳

Changing the expected result to `"\u2233"` seems fine.

Reply:
I see, got it. Added.